open-sphere-plugins/ogc-server/src/main/java/io/opensphere/server/control/AbstractServerSourceController.java

```diff
@@ -361,4 +367,18 @@ public V call()
          */
         protected abstract V callOnce();
     }
+
+    /** Listener for deactivation half of a reload (deactivation + activation) to be finished. */
+    @FunctionalInterface
+    public interface SourceReloadListener
```

Comment:
Could AbstractSeverSourceController just deactivate and reactivate all of its sources then subclasses do not need to implement this.